merge.go

```diff
@@ -261,6 +296,12 @@ func WithTypeCheck(config *Config) {
 	config.TypeCheck = true
 }
 
+// WithSliceDeepCopy will merge slice element one by one with Overwrite flag.
```

Comment:
We should clearly note that it merges the same index. That way the user is informed that the order of the slices is important.